daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/CompiledPackages.scala

```diff
@@ -24,6 +25,7 @@ final class PureCompiledPackages private (
     packages: Map[PackageId, Package],
     defns: Map[SDefinitionRef, SExpr])
     extends CompiledPackages {
+  override def packageIds = packages.keySet
```

Comment:
```suggestion
  override val packageIds = packages.keySet
```
Any reason to recompute every time?

Reply:
`val` is significantly less safe here (there is no warning if you change `CompiledPackages` to make it crash, heck it might already), and `#keySet` is very cheap.